src/catalog/src/lib.rs

```diff
@@ -176,28 +178,56 @@ pub struct BootstrapArgs {
     pub bootstrap_role: Option<String>,
 }
 
-/// A read only API for the durable catalog state.
+/// An API for opening a durable catalog state.
 #[async_trait]
-pub trait ReadOnlyDurableCatalogState: Debug + Send {
-    /// Reports if the catalog state has been initialized.
-    async fn is_initialized(&mut self) -> Result<bool, Error>;
-
-    // TODO(jkosh44) add and implement open methods to be implementation agnostic.
-    /*   /// Checks to see if opening the catalog would be
-    /// successful, without making any durable changes.
+pub trait OpenableDurableCatalogState<D: DurableCatalogState>: Debug + Send {
+    /// Opens the catalog in a mode that accepts and buffers all writes,
+    /// but never durably commits them. This is used to check and see if
+    /// opening the catalog would be successful, without making any durable
+    /// changes.
     ///
     /// Will return an error in the following scenarios:
-    ///   - Catalog not initialized.
+    ///   - Catalog initialization fails.
     ///   - Catalog migrations fail.
-    async fn check_open(&self) -> Result<(), Error>;
+    async fn open_check(
```

Comment:
🚲🚲🚲 bikeshed alert 🚲🚲🚲

When I saw "check" I thought read-only, which this API sort of is, what do you think about just `open_savepoint`? Totally optional and not part of this PR so really up to you

Reply:
I'm not in love with "check" either. I wanted to avoid "savepoint" to avoid tying it to the CRDB implementation, but I think it is probably better than check. For now I've switched back to "savepoint" unless we can come up with something better.